mods/beds/functions.lua

```diff
@@ -224,6 +238,37 @@ minetest.register_on_leaveplayer(function(player)
 	end
 end)
 
+minetest.register_on_dieplayer(function(player)
+	local pos = player:get_pos()
+	local yaw = get_look_yaw(pos)
+
+	lay_down(player, nil, pos, false)
+	player:set_look_horizontal(yaw)
+	player:set_pos(pos)
+
+	minetest.chat_send_all(player:get_player_name().." died.")
```

Comment:
Why is this here?

Reply:
When the sleeping player dies; they are kicked out of the bed by the code on 245. That makes them go to their spawn position but are still laying down. Instead, I made it so that it keeps the player lying on the bed so that they are still in the same position as before they died.

Reply:
I meant the `chat_send_all`

Reply:
I've removed it; I got it from the dev wiki in on the `register_on_dieplayer` page.